internal/service/location/key.go

```diff
@@ -0,0 +1,372 @@
+
+	var plan resourceKeyData
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	in := &location.CreateKeyInput{}
+	resp.Diagnostics.Append(flex.Expand(ctx, plan, in)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// 'ExpireTime' must not be set when 'NoExpiry' has value true.
+	if !plan.NoExpiry.IsNull() {
+		in.NoExpiry = aws.Bool(plan.NoExpiry.ValueBool())
+	} else if !plan.ExpireTime.IsNull() {
+		expireTime, _ := time.Parse(time.RFC3339, plan.ExpireTime.ValueString())
+		in.ExpireTime = aws.Time(expireTime)
+	}
+	in.Tags = getTagsInV2(ctx)
+
+	out, err := conn.CreateKey(ctx, in)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.Location, create.ErrActionCreating, ResNameKey, plan.KeyName.String(), err),
+			err.Error(),
+		)
+		return
+	}
+	if out == nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.Location, create.ErrActionCreating, ResNameKey, plan.KeyName.String(), nil),
+			errors.New("empty output").Error(),
+		)
+		return
+	}
+
+	plan.ID = flex.StringToFramework(ctx, out.KeyName)
+	plan.KeyName = flex.StringToFramework(ctx, out.KeyName)
+
+	// Read after create to get computed attributes omitted from the create response
+	readOut, err := findKeyByID(ctx, conn, plan.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.Location, create.ErrActionSetting, ResNameKey, plan.ID.String(), err),
+			err.Error(),
+		)
+		return
+	}
+	plan.CreateTime = flex.StringValueToFramework(ctx, readOut.CreateTime.Format(time.RFC3339))
+	plan.Key = flex.StringToFramework(ctx, readOut.Key)
+	plan.KeyARN = flex.StringToFrameworkARN(ctx, readOut.KeyArn)
+	plan.UpdateTime = flex.StringValueToFramework(ctx, readOut.UpdateTime.Format(time.RFC3339))
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+func (r *resourceKey) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	conn := r.Meta().LocationClient(ctx)
+
+	var state resourceKeyData
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	out, err := findKeyByID(ctx, conn, state.ID.ValueString())
+	if tfresource.NotFound(err) {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.Location, create.ErrActionSetting, ResNameKey, state.ID.String(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	state.CreateTime = flex.StringValueToFramework(ctx, out.CreateTime.Format(time.RFC3339))
+	state.UpdateTime = flex.StringValueToFramework(ctx, out.UpdateTime.Format(time.RFC3339))
+	if out.ExpireTime == nil {
+		state.NoExpiry = flex.BoolToFramework(ctx, ptr.Bool(true))
+	} else {
+		state.ExpireTime = flex.StringValueToFramework(ctx, out.ExpireTime.Format(time.RFC3339))
+	}
+
+	resp.Diagnostics.Append(flex.Flatten(ctx, out, &state)...)
```

Comment:
It looks like there is some work to include timestamps in the auto-flatteners/expanders but for now I had to add these lines specifically in addition to the `flex.Flatten` catch-all.